x-pack/plugins/lens/public/xy_visualization/xy_config_panel/layer_header.tsx

```diff
@@ -59,8 +59,8 @@ function DataLayerHeader(props: VisualizationLayerWidgetProps<State>) {
   const [isPopoverOpen, setPopoverIsOpen] = useState(false);
   const { state, layerId } = props;
   const layers = state.layers.filter(isDataLayer);
```

Comment:
maybe even I put it in a selector that would return `[layer, id] = getDataLayerById(layerId)`. Probably we can reuse it in future

Reply:
Mmmm not sure. But you've got a point that `layers` is only used here, so perhaps something lighter can be done.

```js
const [layer, index] = state.layers.map( (l, i) => [l, i] ).find( ([layer]) => layer.layerId === layerId );
```

2 iteration instead of 3. WDYT?
TBH I do not think it changes much on the performance side, but if something better can be done I'd take it :)